manager/integration/tests/test_recurring_job.py

```diff
@@ -2152,6 +2152,236 @@ def test_recurring_job_filesystem_trim(client, core_api, batch_v1_api, volume_na
+    volume = wait_for_volume_detached(client, volume.name)
+
+    pv_name = volume_name + "-pv"
+    create_pv_for_volume(client, core_api, volume, pv_name)
+
+    pvc_name = volume_name + "-pvc"
+    create_pvc_for_volume(client, core_api, volume, pvc_name)
+    add_recurring_job_source_to_pvc(pvc_name, core_api)
+
+    recurring_job_name_1 = f'{RECURRING_JOB_NAME}-1'
+    recurring_job_name_2 = f'{RECURRING_JOB_NAME}-2'
+    recurring_jobs = {
+        recurring_job_name_1: {
+            TASK: SNAPSHOT,
+            GROUPS: [],
+            CRON: "*/2 * * * *",
+            RETAIN: 1,
+            CONCURRENCY: 1,
+            LABELS: {},
+        },
+        recurring_job_name_2: {
+            TASK: SNAPSHOT,
+            GROUPS: [recurring_job_name_2],
+            CRON: "*/2 * * * *",
+            RETAIN: 1,
+            CONCURRENCY: 1,
+            LABELS: {},
+        },
+    }
+    create_recurring_jobs(client, recurring_jobs)
+    check_recurring_jobs(client, recurring_jobs)
+
+    claim = core_api.read_namespaced_persistent_volume_claim(
+        name=pvc_name, namespace='default'
+    )
+
+    if claim.metadata.labels is None:
+        claim.metadata.labels = {}
+
+    label_key_1 = f"recurring-job.longhorn.io/{recurring_job_name_1}"
+    label_key_2 = f"recurring-job-group.longhorn.io/{recurring_job_name_2}"
+    claim.metadata.labels[label_key_1] = "enabled"
+    claim.metadata.labels[label_key_2] = "enabled"
+    core_api.replace_namespaced_persistent_volume_claim(
+        pvc_name, 'default', claim
+    )
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[recurring_job_name_1],
+                                         groups=[recurring_job_name_2])
+
+    claim = core_api.read_namespaced_persistent_volume_claim(
+        name=pvc_name, namespace='default'
+    )
+
+    del claim.metadata.labels[label_key_2]
+    core_api.replace_namespaced_persistent_volume_claim(
+        pvc_name, 'default', claim
+    )
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[recurring_job_name_1],
+                                         groups=[])
+
+    recurring_job_1 = client.by_id_recurring_job(recurring_job_name_1)
+    client.delete(recurring_job_1)
+
+    wait_for_volume_recurring_job_update(volume,
+                                         jobs=[],
+                                         groups=[DEFAULT])
+
+    claim = core_api.read_namespaced_persistent_volume_claim(
+        name=pvc_name, namespace='default'
+    )
+
+    if claim.metadata.labels is None:
+        claim.metadata.labels = {}
+
+    unexpected_count = 0
+    for label in claim.metadata.labels:
+        if not label.startswith("recurring-job"):
+            continue
+
+        if label == "recurring-job.longhorn.io/source":
+            continue
+
+        unexpected_count += 1
+    assert unexpected_count == 0
```

Comment:
Why need this check? the labels of the PVC are deleted and volume labels also, so what do you want to check?

Reply:
The purpose of this check is to ensure when the Volume gets updated with the default recurring job (or any recurring job), the changes do not get reflected back to the PVC.